genomic_data_service/region_indexer_encode.py

```diff
@@ -0,0 +1,140 @@
+from genomic_data_service.region_indexer_task import index_file
+from genomic_data_service.region_indexer_elastic_search import RegionIndexerElasticSearch
+
+import requests
+
+from os import environ
+
+
+if 'ES' in environ:
+    es_uri = [environ['ES']]
+else:
+    es_uri = ['localhost']
+es_port = 9201
+
+
+SUPPORTED_CHROMOSOMES = [
+    'chr1', 'chr2', 'chr3', 'chr4', 'chr5', 'chr6', 'chr7', 'chr8', 'chr9',
+    'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15', 'chr16', 'chr17',
+    'chr18', 'chr19', 'chr20', 'chr21', 'chr22', 'chrX', 'chrY'
+]
+
+SEARCH_ENDPOINT = 'https://www.encodeproject.org/search/'
+SUPPORTED_ASSEMBLIES = ['GRCh38', 'mm10']
+ALLOWED_STATUSES = ['released']
+ASSAY_TITLES = ['TF ChIP-seq', 'DNase-seq', 'ATAC-seq', 'eCLIP', 'RAMPAGE', 'CAGE', 'ChIA-PET']
+
+ENCODE_QUERY = [
+    'field=files',
+    'field=biosample_ontology',
+    'field=target.name',
+    'field=biosample_term_name',
+    'field=documents',
+    'field=uuid',
+    'field=assay_term_name',
+    'field=annotation_type',
+    'field=reference_type',
+    'field=collection_type',
+    'files.preferred_default=True',
+    '&'.join([f'status={st}' for st in ALLOWED_STATUSES]),
+    '&'.join([f'assembly={sa}' for sa in SUPPORTED_ASSEMBLIES]),
+    'format=json',
+    'limit=all'
+]
+
+
+def index_encode_files(query, requirements, ignore, dry_run=False):
+    endpoint = f"{SEARCH_ENDPOINT}?{'&'.join(query)}"
+
+    items = requests.get(endpoint).json()['@graph']
+
+    accessions = []
+
+    for item in items:
+        files = item.get('files', [])
+
+        for f in files:
+            if f['accession'] in ignore:
+                continue
+
+            valid = True
+            for requirement in requirements.keys():
+                if f.get(requirement, '') not in requirements[requirement]:
+                    valid = False
+
+            if valid:
+                accessions.append(f['accession'])
+
+                if not dry_run:
+                    index_file.delay(f, item, es_uri, es_port)
+
+    return accessions
+
+
+def encode_annotations(ignore=[], dry_run=False):
```

Comment:
Using mutable default arg like empty list can sometime bite you since it is instantiated once and persists across function calls. Probably not a problem if you never mutate it.